subprojects/enterprise/src/main/java/org/gradle/internal/tasks/OutputFileProperty.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tasks;
+
+public interface OutputFileProperty extends FileProperty {
+
+    TreeType getTreeType();
+
+    enum TreeType {
+        FILE,
+        DIRECTORY
+    }
```

Comment:
No tree type needed for inputs?

Reply:
We could add it here but we (currently) don't need it.